src/D2L.CodeStyle.Analyzers/Immutability/ImmutableDefinitionChecker.cs

```diff
@@ -301,16 +301,38 @@ out diagnostic
 		}
 
 		private readonly struct AssignmentInfo {
-			public AssignmentInfo(
-				bool isInitiailizer,
-				ExpressionSyntax expression
+			private AssignmentInfo(
+				bool isInitializer,
+				ExpressionSyntax expression,
+				ITypeSymbol assignedType
 			) {
-				IsInitializer = isInitiailizer;
+				IsInitializer = isInitializer;
 				Expression = expression;
+				AssignedType = assignedType;
 			}
 
 			public bool IsInitializer { get; }
 			public ExpressionSyntax Expression { get; }
+			public ITypeSymbol AssignedType { get; }
+
+			public static AssignmentInfo Create(
+				SemanticModel model,
+				bool isInitializer,
+				ExpressionSyntax expression
+			) {

```

Comment:
Might not end up with this, ~important change is it now has `AssignedType` exposed